### ISSUE TITLE
Experiment setting of ET BERT

### ISSUE THREAD

**Issue by JizeCao** [aditya2211/transformer-entity-tracking#1]
Since BERT is bidirectional, simply reconstructing the input cannot explicitly inform the model to be entity-centric. How do you apply the entity conditioned framework to BERT?

**Comment by aditya2211** [comment]
We use the pre-trained special tokens `[CLS]` and `[SEP]` to create inputs of the following two forms:

Entity First: `[CLS]` entity `[SEP]` recipe till time step _t_ `[SEP]`
Entity Last: `[CLS]` recipe till time step _t_ `[SEP]` entity `[SEP]`

and then fine-tune using the output of the `[CLS]` token to get P(y|s_{1,...,t},e).

Yes, you are right that being bidirectional there is no notion of entity being seen later after all the process tokens have formed representations (like in GPT entity last models). In the BERT input formations, both the entity and the process token can attend to each other at all time steps (allowing the entity centric representations). 
The two different formations were to experiment with the temporal embedding.

**Comment by JizeCao** [comment]
Yeah. Since the bidirectional transformer can only get the notion of sequence ordering through position embedding, restructure the input order may be really hard to achieve the goal of "entity-centric". That might be the reason for the poor performance of Bert

**Comment by shanestorks** [comment]
To be clear, in the results reported in your paper, is ET_BERT fine-tuned on inputs of both of these forms? Or did you choose one form which worked better on its own?

**Comment by aditya2211** [comment]
Yes, we experiment with both input formulations for BERT and found the entity first variant performing better. However, being bidirectional, in both the formulations the process and entity can attend to each other (only with a difference in the temporal ordering). Thus, we observed minor performance difference between the two forms.

**Comment by shanestorks** [comment]
Understood, thanks!
